agent/requestor-sdk/src/requestor.rs

```diff
@@ -349,22 +370,37 @@ impl Requestor {
                 break;
             }
 
-            let mut res = activity_api
+            all_res = activity_api
```

Comment:
I think the `let` keyword should stay here and the `all_res` defined outside of the loop should be deleted. Unless there's anything in it that must survive from one iteration to next.

Reply:
See line 394